src/terminal/adapter/DispatchTypes.hpp

```diff
@@ -6,6 +6,7 @@
 namespace Microsoft::Console::VirtualTerminal
 {
     using VTInt = int32_t;
+    using MidiDuration = std::chrono::milliseconds;

```

Comment:
Nit: I think the duration parameters would be better left as they were with the original microsecond precision, and then use an explicit `duration_cast` to get the milliseconds when you need them. Maybe one day we might even be able to use the exact duration value for the wait. As it is, we're left with a `count()` of an unknown scale, which is kind of meaningless. 

Also having to use `::Microsoft::Console::VirtualTerminal::MidiDuration` all over the place is bit painful.

Reply:
I don't mind changing back to `microsecond` at all, but generally I'd prefer having the `MidiDuration` type alias. It makes not just discovering where this type is used easier, but also what the actual type is - similar to `VTInt`.

Reply:
I can understand that for something like `VTInt`, where changing from say `size_t` to `int` is still semantically the same thing, but that's not the case here. You're not really changing the type - you're changing the scale. While `size_t 123` and `int 123` are essentially equivalent, `123 milliseconds` and `123 microseconds` are definitely not!

Reply:
I should add that I wouldn't mind so much if you included a `duration_cast` where you're doing the `count()`, because that's mitigates the risk of the code breaking if someone redefines the `MidiDuration` type.

Reply:
> I can understand that for something like `VTInt`, where changing from say `size_t` to `int` is still semantically the same thing, but that's not the case here. You're not really changing the type - you're changing the scale. While `size_t 123` and `int 123` are essentially equivalent, `123 milliseconds` and `123 microseconds` are definitely not!

I'm fine with reverting all those parts of the PR. 🙂
But while I do that I do have a question: We can represent the DECPS duration as a custom
```cpp
std::chrono::duration<VTInt, std::ratio<1, 32>>
```
...but should we really have all those functions spell out that type every time? Wouldn't a `DECPSDuration` (or `MidiDuration`) alias better?

> I should add that I wouldn't mind so much if you included a `duration_cast` where you're doing the `count()`, because that's mitigates the risk of the code breaking if someone redefines the `MidiDuration` type.

I actually intentionally used an explicit `std::chrono::milliseconds` as the parameter type for the `MidiAudio` class because of that. Even when I revert those changes now I'd like to keep that part as I feel like the `MidiAudio` class isn't really part of the VT system and so it doesn't have to use the same parameter type IMO... I guess? 🤔

Reply:
(James' explanation for why he wants MIDI durations to be separate from DECPS durations is sound to me; I suspect you just haven't seen it yet!)

Reply:
> I actually intentionally used an explicit `std::chrono::milliseconds` as the parameter type for the `MidiAudio` class because of that.

Oh sorry. I completely missed that. The lack of cast is fine then. I still don't like the `MidiDuration`, but maybe that's just a personal preference.

> `MidiAudio` class isn't really part of the VT system and so it doesn't have to use the same parameter type IMO

I agree completely. But as I said below, I chose microseconds for the type because I thought it was a reasonable scale that would cover most use cases. The fact that the current wait operation takes milliseconds is more of an implementation detail and shouldn't really be deciding the API parameters. This is not a big deal though. I just wanted to make my case. I'll leave it up to you now.